core/unit_test/TestGraph.hpp

```diff
@@ -864,6 +900,118 @@ TEST(TEST_CATEGORY, when_all_type) {
   static_assert(std::is_same_v<decltype(tail), node_ref_tail_t>);
 }
 
+#if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP)
+template <GraphNodeType value, typename T>
+__global__ void set_to(T* const data) {
+  data[threadIdx.y] += static_cast<T>(value);
+}
+#endif
+
+template <typename Exec, typename ViewType>
+struct ExternalCapture {
+  template <typename N, typename E, typename T>
+  static auto add(const N&, const E&, const T&) {
+    Kokkos::abort("Not implemented.");
+  }
+};
+
+#if defined(KOKKOS_ENABLE_CUDA)
+template <typename ViewType>
+struct ExternalCapture<Kokkos::Cuda, ViewType> {
+  static_assert(ViewType::rank() == 0);
+
+  static void compute(const Kokkos::Cuda& exec, const ViewType& data) {
+    set_to<GraphNodeType::CAPTURE>
+        <<<dim3(1, 1, 1), dim3(1, 1, 1), 0, exec.cuda_stream()>>>(data.data());
+  }
+
+  template <typename N, typename T>
+  static auto add(const N& node, const Kokkos::Cuda& exec, const T& data) {
+    return node.cuda_capture(exec, [data = data](const Kokkos::Cuda& exec_) {
+      ExternalCapture::compute(exec_, data);
+    });
+  }
```

Comment:
The first argument could also be named `predecessor` or `pred`. Just to make it clear that the result of the capture will be used to create a new node that will follow what we're passing as argument to this function.

Reply:
Done :heavy_check_mark: